Compilation Failure on the Latest Rust Nightly Compiler

UEFI crate version: `0.9.0`

Rust compiler metadata: `rustc --version --verbose`
```
rustc 1.54.0-nightly (4e3e6db01 2021-05-18)
binary: rustc
commit-hash: 4e3e6db011c5b482d2bef8ba02274657f93b5e0d
commit-date: 2021-05-18
host: x86_64-pc-windows-msvc
release: 1.54.0-nightly
LLVM version: 12.0.1
```

Errors:
```
error[E0437]: type `Ok` is not a member of trait `ops::Try`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:175:5
175 |     type Ok = Completion<()>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `ops::Try`

error[E0437]: type `Error` is not a member of trait `ops::Try`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:176:5
    |
176 |     type Error = Error<()>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `ops::Try`

error[E0407]: method `into_result` is not a member of trait `ops::Try`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:178:5
    |
178 | /     fn into_result(self) -> Result<(), ()> {
179 | |         self.into()
180 | |     }
    | |_____^ not a member of trait `ops::Try`

error[E0407]: method `from_error` is not a member of trait `ops::Try`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:182:5
    |
182 | /     fn from_error(error: Self::Error) -> Self {
183 | |         error.status()
184 | |     }
    | |_____^ not a member of trait `ops::Try`

error[E0407]: method `from_ok` is not a member of trait `ops::Try`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:186:5
    |
186 | /     fn from_ok(ok: Self::Ok) -> Self {
187 | |         ok.status()
188 | |     }
    | |_____^ not a member of trait `ops::Try`

error[E0658]: use of unstable library feature 'try_trait_v2'
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:174:6
    |
174 | impl ops::Try for Status {
    |      ^^^^^^^^
    |
    = note: see issue #84277 <https://github.com/rust-lang/rust/issues/84277> for more information
    = help: add `#![feature(try_trait_v2)]` to the crate attributes to enable

error[E0220]: associated type `Error` not found for `Self`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:182:32
    |
182 |     fn from_error(error: Self::Error) -> Self {
    |                                ^^^^^ associated type `Error` not found

error[E0220]: associated type `Ok` not found for `Self`
   --> <path redacted>\uefi-0.9.0\src\result\status.rs:186:26
    |
186 |     fn from_ok(ok: Self::Ok) -> Self {
    |                          ^^ associated type `Ok` not found

error: aborting due to 8 previous errors
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -25,7 +25,8 @@
 
 #![cfg_attr(feature = "exts", feature(allocator_api, alloc_layout_extra))]
 #![feature(auto_traits)]
-#![feature(try_trait)]
+#![feature(control_flow_enum)]
+#![feature(try_trait_v2)]
 #![feature(abi_efiapi)]
 #![feature(negative_impls)]
 #![feature(const_panic)]

--- a/src/result/status.rs
+++ b/src/result/status.rs
@@ -1,6 +1,6 @@
 use super::{Completion, Error, Result};
-use core::fmt::Debug;
-use core::ops;
+use core::{fmt::Debug, num::NonZeroUsize};
+use core::ops::{ControlFlow, FromResidual, Try};
 
 /// Bit indicating that an UEFI status code is an error
 const ERROR_BIT: usize = 1 << (core::mem::size_of::<usize>() * 8 - 1);
@@ -162,8 +162,7 @@ impl Status {
     }
 }
 
-// An UEFI status is equivalent to a Result with no data or rerror payload
-
+// An UEFI status is equivalent to a Result with no data or error payload
 impl From<Status> for Result<(), ()> {
     #[inline]
     fn from(status: Status) -> Result<(), ()> {
@@ -171,20 +170,27 @@ impl From<Status> for Result<(), ()> {
     }
 }
 
-impl ops::Try for Status {
-    type Ok = Completion<()>;
-    type Error = Error<()>;
+pub struct StatusResidual(NonZeroUsize);
 
-    fn into_result(self) -> Result<(), ()> {
-        self.into()
+impl Try for Status {
+    type Output = Completion<()>;
+    type Residual = StatusResidual;
+
+    fn branch(self) -> ControlFlow<Self::Residual, Self::Output> {
+        match NonZeroUsize::new(self.0) {
+            Some(r) => ControlFlow::Break(StatusResidual(r)),
+            None => ControlFlow::Continue(Completion::from(self)),
+        }
     }
 
-    fn from_error(error: Self::Error) -> Self {
-        error.status()
+    fn from_output(output: Self::Output) -> Self {
+        output.status()
     }
+}
 
-    fn from_ok(ok: Self::Ok) -> Self {
-        ok.status()
+impl FromResidual for Status {
+    fn from_residual(r: StatusResidual) -> Self {
+        Status(r.0.into())
     }
 }
 

--- a/src/result/status.rs
+++ b/src/result/status.rs
@@ -1,6 +1,6 @@
 use super::{Completion, Error, Result};
-use core::{fmt::Debug, num::NonZeroUsize};
 use core::ops::{ControlFlow, FromResidual, Try};
+use core::{fmt::Debug, num::NonZeroUsize};
 
 /// Bit indicating that an UEFI status code is an error
 const ERROR_BIT: usize = 1 << (core::mem::size_of::<usize>() * 8 - 1);

--- a/uefi-test-runner/src/proto/pi/mp.rs
+++ b/uefi-test-runner/src/proto/pi/mp.rs
@@ -56,14 +56,14 @@ fn test_get_processor_info(mps: &MpServices) {
     assert_eq!(cpu2.processor_id, 2);
 
     // Check that only CPU 0 is BSP
-    assert_eq!(cpu0.is_bsp(), true);
-    assert_eq!(cpu1.is_bsp(), false);
-    assert_eq!(cpu2.is_bsp(), false);
+    assert!(cpu0.is_bsp());
+    assert!(!cpu1.is_bsp());
+    assert!(!cpu2.is_bsp());
 
     // Check that only the second CPU is disabled
-    assert_eq!(cpu0.is_enabled(), true);
-    assert_eq!(cpu1.is_enabled(), false);
-    assert_eq!(cpu2.is_enabled(), true);
+    assert!(cpu0.is_enabled());
+    assert!(!cpu1.is_enabled());
+    assert!(cpu2.is_enabled());
 
     // Enable second CPU back
     mps.enable_disable_ap(1, true, None).unwrap().unwrap();
@@ -138,12 +138,12 @@ fn test_enable_disable_ap(mps: &MpServices) {
         .unwrap()
         .unwrap();
     let cpu1 = mps.get_processor_info(1).unwrap().unwrap();
-    assert_eq!(cpu1.is_healthy(), false);
+    assert!(!cpu1.is_healthy());
 
     // Mark second CPU as healthy again and check it's status
     mps.enable_disable_ap(1, true, Some(true)).unwrap().unwrap();
     let cpu1 = mps.get_processor_info(1).unwrap().unwrap();
-    assert_eq!(cpu1.is_healthy(), true);
+    assert!(cpu1.is_healthy());
 }
 
 fn test_switch_bsp_and_who_am_i(mps: &MpServices) {

